### ISSUE TITLE
The o1 and o2 attributes of the hppfcl.Contact class return objects at different addresses in Python

### ISSUE THREAD

**Issue by jongyaoY** [humanoid-path-planner/hpp-fcl#590]
I am trying to retrieve the collision object names from the collision results by querying a dictionary created as:

```python
self._names={}
geom = fcl.BVHModelOBB()
geom.beginModel(len(mesh.faces),
                    len(mesh.vertices))
geom.addSubModel(verts, faces)
geom.endModel()
self._names[id(geom)] = name
```
after I get the collision results from the collision manager, I would like to query self._names for the mesh names:

```python
self._manager = fcl.DynamicAABBTreeCollisionManager()
self._manager.setup()

...register objects...

callback = fcl.CollisionCallBackDefault()
self._manager.collide(callback)
contacts = callback.data.result.getContacts()
for contact in contacts:
    names = (self._name[id(contact.o1)],
                  self._name[id(contact.o1)])

```
and I found out `id(contact.o1)` and `id(contact.o2)` change every time I query them:

```python
print(id(contact.o1)) #>>>139658153979680
print(id(contact.o1)) #>>>139658153977104
```
which makes it impossible to retrieve the mesh names

**Comment by jcarpent** [comment]
This behavior is expected, as `contact.o1` and `contact.o2` both refer to C++ pointers.
We will add helpers to extract a unique id, related to the memory address of the C++ pointee.

**Comment by jongyaoY** [comment]
> This behavior is expected, as `contact.o1` and `contact.o2` both refer to C++ pointers. We will add helpers to extract a unique id, related to the memory address of the C++ pointee.

Thanks, then what is the current way of retrieving which objects in the manager that o1 and o2 are referring to?

**Comment by jcarpent** [comment]
Are you able to install hpp-fcl from source?

**Comment by jongyaoY** [comment]
> Are you able to install hpp-fcl from source?
Yes, I complied the package from source. But it didn't resolve the issue. I figured the issue comes from eigenpy, and I noticed in the [CMakeLists.txt](https://github.com/humanoid-path-planner/hpp-fcl/blob/devel/CMakeLists.txt), it's still using the old version of eigenpy. How can I compile it with the latest eigenpy release?

**Comment by jcarpent** [comment]
Please install with conda from a clean env.

**Comment by jongyaoY** [comment]
> Please install with conda from a clean env.

Is there way to compile with cmake and still get the latest eigenpy? since conda takes forever to install

**Comment by nim65s** [comment]
Yes, you can just use standard CMake procedure to configure, build, test and install eigenpy from https://github.com/stack-of-tasks/eigenpy

**Comment by jongyaoY** [comment]
> Yes, you can just use standard CMake procedure to configure, build, test and install eigenpy from https://github.com/stack-of-tasks/eigenpy

I manage to make the python wheel with cmeel and the latest eigenpy(3.8.0). However `id(contact.o1)` still gives me inconsistent ids

**Comment by jongyaoY** [comment]
> Yes, you can just use standard CMake procedure to configure, build, test and install eigenpy from https://github.com/stack-of-tasks/eigenpy

I managed to make the python wheel with cmeel and eigenpy(3.8.0), but `id(contact.o1)` still gives me the inconsistent results

**Comment by nim65s** [comment]
@jongyaoY : it is expected to be used with `contact.o1.id()` I think

**Comment by jongyaoY** [comment]
> @jongyaoY : it is expected to be used with `contact.o1.id()` I think

 `contact.o1.id()`  raises `AttributeError: 'BVHModelOBB' object has no attribute 'id'`

**Comment by jongyaoY** [comment]
I managed to solve the issue by adding one line in [collision-geometries.cc](https://github.com/humanoid-path-planner/hpp-fcl/blob/devel/python/collision-geometries.cc):
```cpp
  #include <eigenpy/id.hpp>
  ...
  class_<BVH, bases<BVHModelBase>, shared_ptr<BVH>>(
      type_name.c_str(), doxygen::class_doc<BVH>(), no_init)
      .def(dv::init<BVH>())
      .def(dv::init<BVH, const BVH&>())
      .DEF_CLASS_FUNC(BVH, getNumBVs)
      .DEF_CLASS_FUNC(BVH, makeParentRelative)
      .DEF_CLASS_FUNC(BVHModelBase, memUsage)
      .def("clone", &BVH::clone, doxygen::member_func_doc(&BVH::clone),
           return_value_policy<manage_new_object>())
      .def_pickle(PickleObject<BVH>())
      .def(SerializableVisitor<BVH>())
      .def(eigenpy::IdVisitor<BVH>());
```

now `contact.o1.id()` gives me what I need

**Comment by jcarpent** [comment]
Solved via #618
